src/vivarium_public_health/metrics/stratification.py

```diff
@@ -0,0 +1,319 @@
+from dataclasses import dataclass
+from enum import Enum
+import itertools
+from typing import Callable, Dict, Iterable, List, Set, Tuple, Union
+
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import PopulationView
+
+
+class SourceType(Enum):
+    COLUMN = "column"
+    PIPELINE = "pipeline"
+    CLOCK = "clock"
+
+
+@dataclass
+class Source:
+    # todo add docstrings
+    name: str
+    type: SourceType
+
+
+@dataclass
+class StratificationLevel:
+    # todo add docstrings
```

Comment:
It would be nice to have the docstrings in this PR

Reply:
I'll add them in a separate PR, if that's alright.